src/Stratis.Bitcoin/Utilities/Extensions/IPExtensions.cs

```diff
@@ -19,7 +19,7 @@ public static IPEndPoint MapToIpv6(this IPEndPoint endPointv4)
         /// <summary>Match the end point with another by IP and port.</summary>
         public static bool Match(this IPEndPoint endPoint, IPEndPoint matchWith)
         {
-            return endPoint.Address.ToString() == matchWith.MapToIpv6().Address.ToString() && endPoint.Port == matchWith.MapToIpv6().Port;
+            return endPoint.MapToIpv6().Address.ToString() == matchWith.MapToIpv6().Address.ToString() && endPoint.Port == matchWith.Port;
```

Comment:
I wonder if there was a reason this wasn't already like this with the Map to Ipv6 on both sides?

Reply:
Yes we (@monsieurleberre) wasn't too sure.  Anyhow it has to be on both sides to match.